pyrobuf/parse_proto.py

```diff
@@ -209,17 +222,19 @@ def _parse_message(self, s, current, tokens, enums, imported_enums):
 
         for token in tokens:
             if token.token_type == 'MESSAGE':
-                current.messages[token.name] = self._parse_message(s, token, tokens, enums, imported_enums)
+                token.full_name = current.full_name + token.name
+                current.messages[token.name] = self._parse_message(s, token, tokens, messages, enums, imported_enums)
+                # updates the dictionary of known/parsed messages.
+                messages.update(current.messages)
```

Comment:
Actually, the one case I gave, where `A` and `B` both have a child named `C`, does work. I think the other issues stand, but perhaps are worthy of their own PRs.